pilot/pkg/proxy/envoy/v1/externalservice.go

```diff
@@ -188,7 +188,7 @@ func buildExternalServiceVirtualHost(serviceName string, externalService *networ
 
 	// FIXME: clusters generated if the routing rule routes traffic to other services will be constructed incorrectly
 	// FIXME: similarly, routing rules for other services that route to this external service will be constructed incorrectly
-	routes := buildDestinationHTTPRoutes(node, service, port, proxyInstances, config, buildClusterFunc)
+	routes := buildDestinationHTTPRoutes(node, service, port, proxyInstances, config, false, buildClusterFunc)
```

Comment:
it would be more readable to have separate V1 and V2 functions (as wrappers, or defined in terms of each other). here are throughout.